### PR TITLE
[HostMetrics Receiver] Add support for process state

### PR DESCRIPTION
**Pull request by ishleenk17**
The Hostmetrics receiver doesn't report the process state in the process scraper.
This PR adds the process state to the Resource Attributes. 

Output would look like below: 

> ResourceMetrics #544
> Resource SchemaURL: https://opentelemetry.io/schemas/1.9.0
> Resource attributes:
>      -> process.pid: Int(99349)
>      -> process.parent_pid: Int(1)
>      -> process.executable.name: Str(XProtect)
>      -> process.executable.path: Str(/Library/Apple/System/Library/CoreServices/XProtect.app/Contents/MacOS/XProtect)
> `-> process.state: Str(sleep)`
>     -> process.command: Str(/Library/Apple/System/Library/CoreServices/XProtect.app/Contents/MacOS/XProtect)
>      -> process.command_line: Str(/Library/Apple/System/Library/CoreServices/XProtect.app/Contents/MacOS/XProtect)
>      -> process.owner: Str(ishleenkaur)
>      -> host.name: Str(ishleens-laptop.local)
>      -> os.type: Str(darwin)
> ScopeMetrics #0
> ScopeMetrics SchemaURL: 
> InstrumentationScope otelcol/hostmetricsreceiver/process 1.0.0

### CODE REVIEW

**Comment by braydonk** [comment]
Thanks for the PR!

I don't think this will work as a resource attribute. The process resource needs to be immutable for an instance of a process. Otherwise, the resource will be interpreted as a new resource and change the meaning of each timeseries datapoint it's associated with.

I think `process.status` would make more sense as a metric. I've opened an issue for the System Semantic Conventions Working Group to come up with a design for this to inform the instrumentation in the hostmetricsreceiver.

**Comment by ishleenk17** [comment]
@braydonk: Can we please review this?

**Comment by ChrsMark** [comment]
I agree with what @braydonk mentioned at https://github.com/open-telemetry/semantic-conventions/issues/1181#issuecomment-2196663949, that we should first ship the SemConv part before moving on with the implementation.

**Comment by braydonk** [comment]
Part of this PR will need to be redone with the proper code generation procedure.

1. Add `process.status` to [metadata.yaml in the `resource_attributes` section](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1724b6dd34c4715d4067f5c097d93aa6f519947e/receiver/hostmetricsreceiver/internal/scraper/processscraper/metadata.yaml#L8). Set `enabled` to `false`.
2. In the root of the repo, run `make generate`. This will generate new code in `processscraper/internal/metadata`, including code to set the `process.status` resource attribute.
3. Use that generated code in this PR.

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go [hunk 1]
```diff
@@ -253,6 +253,10 @@ func (s *scraper) getProcessMetadata() ([]*processMetadata, error) {
 				errs.AddPartial(0, fmt.Errorf("error reading username for process %q (pid %v): %w", executable.name, pid, err))
 			}
 		}
+		status, err := getProcessStatus(ctx, handle)
+		if err != nil {
+			errs.AddPartial(1, fmt.Errorf("error reading process cgroup for pid %v: %w", pid, err))
```
Comment (braydonk) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper.go:258]:
```suggestion
			errs.AddPartial(1, fmt.Errorf("error reading process status for pid %v: %w", pid, err))
```

**Comment by ishleenk17** [comment]
> Part of this PR will need to be redone with the proper code generation procedure.
> 
> 1. Add `process.status` to [metadata.yaml in the `resource_attributes` section](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1724b6dd34c4715d4067f5c097d93aa6f519947e/receiver/hostmetricsreceiver/internal/scraper/processscraper/metadata.yaml#L8). Set `enabled` to `false`.
> 2. In the root of the repo, run `make generate`. This will generate new code in `processscraper/internal/metadata`, including code to set the `process.status` resource attribute.
> 3. Use that generated code in this PR.

Thanks @braydonk , I'll make the needed changes. 
Is there any conclusion on the semconv change post which this can be merged?

**Comment by ishleenk17** [comment]
> Is there any conclusion on the semconv change post which this can be merged?

@braydonk

**Comment by ChrsMark** [comment]
@ishleenk17 the definition of the respective SemConv is blocked at https://github.com/open-telemetry/semantic-conventions/issues/1181 (you can check the details there).
In this, I'm not sure if we are ok with proceeding with the addition in the Collector without a clear resolution in the SemConv first. Since you are interested, would you mind raising this need in the respective SemConv's issue?

@open-telemetry/collector-contrib-approvers is there any generic guidance for such cases? Or it's up to the component's code-owners to decide?

**Comment by mx-psi** [comment]
I don't think we have generic guidance about this, it would be good to add some (if you/somebody else is up for it I am happy to review).

My personal view is that it may be okay to add these under an alpha feature gate at the discretion of codeowners.

**Comment by ChrsMark** [comment]
> I don't think we have generic guidance about this, it would be good to add some (if you/somebody else is up for it I am happy to review).
> 
> My personal view is that it may be okay to add these under an alpha feature gate at the discretion of codeowners.

The feature gate approach sounds reasonable.
I think we can see if the feature gate approach will work well in this specific case and then we could use it as reference for future issues/cases.

**Pull request closed without merging**